core/hash_map.h

```diff
@@ -74,13 +78,14 @@ class HashMapHahserDefault {
  * @param TKey  Key, search is based on it, needs to be hasheable. It is unique in this container.
  * @param TData Data, data associated with the key
  * @param Hasher Hasher object, needs to provide a valid static hash function for TKey
+ * @param Comparator comparator object, needs to be able to safely compare two TKey values
```

Comment:
Yes, exactly. Go randomizes the key, Godot normalizes nan values. It doesn't say anywhere 'oh, btw, if you use a (compound) key with a nan value we produce garbage and you won't be able to retrieve by key any longer. Which is the type of comment you apparently want.

I'm saying, I already had what you were asking for.